x/bank/keeper/view.go

```diff
@@ -259,3 +289,56 @@ func UnmarshalBalanceCompat(cdc codec.BinaryCodec, bz []byte, denom string) (sdk
 
 	return sdk.NewCoin(denom, amount), nil
 }
+
+// lockedCoinsGetter is a struct that houses a GetLockedCoinsFn.
+// It exists so that the GetLockedCoinsFn can be updated in the ViewKeeper without needing to have a pointer receiver.
+type lockedCoinsGetter struct {
```

Comment:
Nitpick, but is there a better name rather than lockedCoinsGetter? Some suggestions would be lockedCoinsPipeline, lockedCoinsChains, etc...

Reply:
> There are two major problems in Computer Science:
> 1. Naming things.
> 2. Cache invalidation.
> 3. Off-by-one errors.

It's not a pipeline or chain though, it's just a thing that holds a single `GetLockedCoinsFn`. The pipeline/chaining part is handled through function composition managed by the `GetLockedCoinsFn` itself.

With the send restrictions, the function is named `SendRestrictionFn`, its holder for the keeper is `sendRestriction`, and its function for calling it is `apply`. But for this, I didn't want `apply` since it's supposed to be just a lookup, so I settled on `getLockedCoins`. I then didn't want to call the holder `getLockedCoins` because its too action sounding for an object name, and it'd then have the same name as its primary function; the `LockedCoins` function would have the line `return k.getLockedCoins.getLockedCoins(ctx, addr)`.

Whenever I find myself wanting to name something "getter", I try to come up with something else that makes sense and is similarly succinct. In this case, I'm looking to name, "A thing that holds a `GetLockedCoinsFn` and has some methods for managing and using it." I thought about `getLockedCoinsFnHolder`, but a) that's longer than I was hoping for, and b) I dislike "holder" in names more than I dislike "getter". I also considered "getLockedCoinsFnWrapper", but ultimately decided I liked the short length of `lockedCoinsGetter` better. That worked out too since I later realized I needed to put a safety net around each `GetLockedCoinsFn` that prevents it from returning negative coin values, and ended up calling that `getLockedCoinsFnWrapper`; I'm not too happy about that name either, though.

I also considered names with `bin`, `bag`, `container`, `case`, `casing`, `pointer`, `reference`, and `manager`. I  also considered calling it `getLockedCoinsFn`, but thought that'd be too confusing.

I also considered just calling it `getLockedCoinsFn`, `lockedCoinsFn`, or `lockedCoins`, and calling the `getLockedCoins` method just `get`. I thought `getLockedCoinsFn` was too easily confused with `GetLockedCoinsFn`. I'm hesitant on `lockedCoinsFn` because that doesn't convey that all it's concerned with is looking up locked coins. And I thought `lockedCoins` was too easily confused with the `LockedCoins` method. However, since the only place this is applied is in `LockedCoins`, maybe it does make sense to just call this struct `lockedCoins`, and call its method `get`. I didn't do that originally because I wanted it clear that it's just a getter.

Ultimately, I decided that, since it's private, I could live with calling it `lockedCoinsGetter`.

I do think I'll rename `getLockedCoinsFnWrapper` to something like `onlyPositiveCoins` though.

Reply:
The reason I visualized this object as multi step processor is because of the append, prepend, and clear functions. Your choice makes sense if you are trying to keep the naming consistent with the rest of the code.